pkcs11/util/ec.py

```diff
@@ -11,10 +11,18 @@
 )
 from asn1crypto.algos import DSASignature
 from asn1crypto.core import OctetString
-
 from ..constants import Attribute, ObjectClass
 from ..mechanisms import KeyType
 
+# patch in RFC 8410 (https://tools.ietf.org/html/rfc8410)
+rfc_8410_oids = {
+    "X25519": "1.3.101.110",
+    "X448": "1.3.101.111",
+    "Ed25519": "1.3.101.112",
+    "Ed448": "1.3.101.113",
+}
+NamedCurve._map.update({v: k for k, v in rfc_8410_oids.items()})
```

Comment:
I'm happy with everything here but patching `asn1crypto`. You can pass in OIDs directly as strings. Which I think is a good compromise until `asn1crypto` is updated.

Reply:
Ah interesting, that was not obvious. Agreed, hope it gets merged; there are a bunch of open PRs.